UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/examresult/ExamResultContractUtil.java

```diff
@@ -119,6 +124,14 @@ public void checkParamsAddExamResult(Context ctx, String[] params) throws Parame
         if (distinctExamIds(examResultEntries)) {
             invalidParams.add(getDistinctExamIdsParam());
         }
+
+        ExamContractUtil cUtil = new ExamContractUtil();
+        Exam exam = cUtil.getState(ctx.getStub(), examResultEntries.get(0).getExamId(), Exam.class);
+
+        if (Instant.now().isBefore(exam.getDate())) {
+            invalidParams.add(new InvalidParameter().name("examId").reason("The exam did not happen, yet."));
```

Comment:
To be consistent, this invalid parameter should be defined in its own method so it is defined in one place in case the same invalid parameter can happen in other places. Since I don't currently see where this could happen, this is also fine.